app/src/pages/p/[projectSlug]/usage/index.tsx

```diff
@@ -16,72 +16,240 @@ import {
   Tr,
   VStack,
   Box,
+  Spacer,
+  Tabs,
+  TabList,
+  Tab,
+  TabPanels,
+  TabPanel,
 } from "@chakra-ui/react";
-import { DollarSign, Hash } from "lucide-react";
+import { ChevronRightIcon, ChevronLeftIcon, DollarSign, Hash } from "lucide-react";
+import { useRouter } from "next/router";
+import { useEffect, useMemo } from "react";
+import { useQueryParam, StringParam } from "use-query-params";
 import { ProjectLink } from "~/components/ProjectLink";
+import CostGraph from "~/components/dashboard/CostGraph";
 
 import UsageGraph from "~/components/dashboard/UsageGraph";
 import AppShell from "~/components/nav/AppShell";
 import { modelInfo } from "~/server/fineTuningProviders/supportedModels";
-import { api } from "~/utils/api";
-import dayjs from "~/utils/dayjs";
-import { useSelectedProject } from "~/utils/hooks";
+import dayjs, { formatToUTCDayMonth } from "~/utils/dayjs";
+import { useSelectedProject, useStats } from "~/utils/hooks";
 
 const numberWithDefault = (num: number | string | bigint | null, defaultValue = 0) =>
   Number(num ?? defaultValue);
 
 export default function Usage() {
   const { data: selectedProject } = useSelectedProject();
 
-  const stats = api.usage.stats.useQuery(
-    { projectId: selectedProject?.id ?? "" },
-    { enabled: !!selectedProject },
-  );
+  const router = useRouter();
+
+  const [startDate, setStartDate] = useQueryParam("start", StringParam);
```

Comment:
Since these are generally updated at the same time, you'd likely find it more convenient to use the `useQueryParams` hook from this library, which lets you set multiple pieces of state simultaneously. That also would let you avoid changing the batching behavior (which is probably fine to change, but unnecessary in this case).